### PR TITLE
[batch] various improvements

### LINKED ISSUE #7756: [batch] hailctl batch list broken
I can't use `hailctl batch list`
```
Last login: Wed Dec 18 15:05:20 on ttys001
# hailctl batch list
Traceback (most recent call last):
  File "/usr/local/bin/hailctl", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/hailtop/hailctl/__main__.py", line 103, in main
    cli.main(args)
  File "/usr/local/lib/python3.7/site-packages/hailtop/hailctl/batch/cli.py", line 97, in main
    jmp[args.module].main(args, pass_through_args, client)
  File "/usr/local/lib/python3.7/site-packages/hailtop/hailctl/batch/list_batches.py", line 39, in main
    batch_list = client.list_batches(success=success, complete=complete, attributes=attributes)
TypeError: list_batches() got an unexpected keyword argument 'success'
# 
```

### PR DESCRIPTION
**Pull request by danking**
- retry every deadlock in two deadlock prone SQL operations
- add prometheus metrics for cores
- fix prometheus when you're not in the default namespace
- retry every docker 500 error, it's 500, not our fault, just retry, right?
- create a billing account for the dev deploying user
- rewrite a couple queries to harmonize table locking a bit
- add globals to delete tables script
- fix list_batches, which was broken by the query language changes
- include primary services developers' namespaces in prometheus monitoring

Fixes #7756

### CODE REVIEW

**Comment by cseed** [comment]
> retry every deadlock in two deadlock prone SQL operations

I prefer @jigold's change which is almost ready: https://github.com/hail-is/hail/pull/7782

 > retry every docker 500 error, it's 500, not our fault, just retry, right?

500 is meaningless, docker returns 500 for everything.  I'm quite concerned about infinite retry loops here, and I'd rather have documentation on the errors we're getting from docker if possible.

Looking over the other changes.

**Comment by danking** [comment]
> 500 is meaningless, docker returns 500 for everything. I'm quite concerned about infinite retry loops here, and I'd rather have documentation on the errors we're getting from docker if possible.

OK, I can add the two other failures I found, but it's infuriating to get most of the way through 100k jobs and then have one fail for some new 500. It feels like there's a long tail of rare docker errors.

File: batch/create-batch-tables-pod.yaml [hunk 1]
```diff
@@ -20,6 +20,15 @@ spec:
...
          mysql --defaults-extra-file=/sql-config/sql-config.cnf < ./create-billing-projects.sql
+         {% if scope == 'dev' %}
+         mysql --defaults-extra-file=/sql-config/sql-config.cnf <<'EOF'
+       INSERT INTO `billing_projects` (`name`)
+       VALUES ("hail");
+
+       INSERT INTO `billing_project_users` (`billing_project`, `user`)
+       VALUES ("hail", "{{ default_ns.name }}");
+       EOF
+         {% endif %}
```
Comment (cseed) [inline batch/create-batch-tables-pod.yaml]: This is about to go away when my database migrations PR goes in. Can you remove it from here and I'll add it there?  This doesn't feel like quite the right place for this, either, but I don't know where it should go.

File: batch/create-batch-tables.sql [hunk 2]
```diff
@@ -173,13 +173,13 @@ BEGIN
 
   SET msec_mcpu_diff = msec_diff * job_cores_mcpu;
 
-  UPDATE batches
```
Comment (cseed) [inline batch/create-batch-tables.sql]: Can you add a comment about the organizing principle somewhere?  It isn't immediately obvious to me.

**Comment by cseed** [comment]
I agree.  And parsing strings to figure out what's going on is insanity.  Here's a middle-ground: retry unknown 500 errors once.  If they really are rare, they won't reoccur (p^2 very small).

File: batch/create-batch-tables.sql (hunk 2 above)
Comment (danking) [reply batch/create-batch-tables.sql]: OK, I added some information. I admittedly don't have good data on how these changes affect the performance of batch.

**Comment by danking** [comment]
irrelevant now

**Pull request closed without merging**